lib/rules/prefer-core-components-stylesheet.js

```diff
@@ -0,0 +1,42 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Use StyleSheet from core-components instead of react-native',
+    },
+    fixable: 'code',
+  },
+  create(context) {
+    const sourceCode = context.getSourceCode();
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value != 'react-native') return {}
+        const styleSheetImport = node.specifiers.find((specifier) => (specifier.imported !== undefined && specifier.imported.name === 'StyleSheet'));
+        if (!styleSheetImport) return {}
+
+        if (sourceCode.getText(node).match(/^import\s\{\sStyleSheet\s\}\sfrom\s'react-native';$/)) {
```

Comment:
I see this working in the tests - how does this regex work to match it as one of the elements in `{ ... }`? Furthermore, does this work well when imports are on new lines?

Reply:
This is just for the simple case where there is only one import on one line, since in this case you need to delete the whole line. The else covers all other cases. Here is the regex and fix in action: https://regex101.com/r/6dFOHA/1

All error cases are handled correctly, but there is one auto fix case that only partially works. If you have a multi line import, with just one import, you will end up with an empty import.
Example:
```
import {
  StyleSheet
} from 'react-native';
```
Resolves to:
```
import { } from 'react-native';
```

Reply:
I don't think there is a better solution to that - our eslint has some issues with things like this already.